src/Tooltip2.tsx

```diff
@@ -0,0 +1,292 @@
+      }),
+      ...((direction === 's' || direction === 'se' || direction === 'sw') && {
+        top: '100%',
+        right: '50%',
+        marginTop: '6px',
+      }),
+      ...(direction === 'se' && {
+        right: 'auto',
+        left: '50%',
+        marginLeft: `-${TOOLTIP_ARROW_EDGE_OFFSET}px`, // space.3?
+      }),
+      ...(direction === 'sw' && {
+        marginRight: `-${TOOLTIP_ARROW_EDGE_OFFSET}px`, // space.3?
+      }),
+      ...(direction === 'e' && {
+        left: '100%',
+        bottom: '50%',
+        marginLeft: '6px',
+        transform: 'translateY(50%)',
+      }),
+      ...(direction === 'w' && {
+        right: '100%',
+        bottom: '50%',
+        marginRight: '6px',
+        transform: 'translateY(50%)',
+      }),
+      // only for s and n direction to move the popover bovy to the center of the trigger
+      ...((direction === 'n' || direction === 's') && {
+        transform: 'translateX(50%)',
+      }),
+      // Left align tooltips with align prop
+      ...(align === 'left' && {
+        right: '100%',
+        marginLeft: '0',
+      }),
+      // Right align tooltips with align prop
+      ...(align === 'right' && {
+        right: '0',
+        marginRight: '0',
+      }),
+      // Multiline tooltips with wrap prop
+      ...(wrap && {
+        display: 'table-cell',
+        width: 'max-content',
+        maxWidth: '250px',
+        wordWrap: 'break-word',
+        whiteSpace: 'pre-line',
+        borderCollapse: 'separate',
+      }),
+      // Some styles of the directions need to be overriden when wrap is true
+      ...(wrap &&
+        (direction === 's' || direction === 'n') && {
+          right: 'auto',
+          left: '50%',
+          transform: 'translateX(-50%)',
+        }),
+      ...(wrap &&
+        (direction === 'w' || direction === 'e') && {
+          right: '100%',
+        }),
+    }),
+  },
+
+  '&::before, & > span': {
+    ...(open && {
+      display: 'inline-block',
+      textDecoration: 'none',
+      animationName: 'tooltip-appear',
+      animationDuration: '0.1s',
+      animationFillMode: 'forwards',
+      animationTimingFunction: 'ease-in',
+      animationDelay: noDelay ? '0s' : '0.4s',
+    }),
+  },
+})
+
+const Tooltip2: React.FC<React.PropsWithChildren<Tooltip2Props>> = ({
+  direction = 'n',
+  text,
+  noDelay,
+  align,
+  wrap,
+  sx = {},
+  children,
+  type = 'label',
+  ...props
+}) => {
+  const id = useId()
+  const tooltipRef = useRef<HTMLDivElement>(null)
+  const child = Children.only(children) // make sure there is only one child
+  const [open, setOpen] = useState(false)
+
+  // we need this check for every render
+  useEffect(() => {
+    if (tooltipRef.current) {
+      const childNode = tooltipRef.current.children[0] // For now, I assume it has one node but that is not true
+      if (!isFocusable(childNode as HTMLElement)) {
+        throw new Error('Tooltip2: The child element must be focusable')
+      }
```

Comment:
Couple of ideas for the guard block:

```suggestion
  if (__DEV__) {
    useEffect(() => {
      if (tooltipRef.current) {
         // For now, I assume it has one node but that is not true
        const childNode = tooltipRef.current.children[0]
        invariant(
          isFocusable(childNode as HTMLElement),
          'The `Tooltip2` component expects a single React element that contains interactive content. Consider using a `<button>` or equivalent interactive element instead.'
        );
      }
    });
  }
```

Reply:
Question: would it be possible to add a `ref` to the trigger element in `cloneElement`?

Reply:
>Couple of ideas for the guard block

Oh I love that 💖  I forgot we have this utils now - thank you! (I applied suggestion in the local to avoid linting issues)

>Question: would it be possible to add a ref to the trigger element in cloneElement?

I guess we could, where would we use it?